### DISCUSSION TITLE
indentical pub/priv keys from different derivation paths

### DISCUSSION THREAD

**Discussion by ArturLevchuk** [trustwallet/wallet-core#3699] [Q&A] [answered]
maybe it's problem with my code but i get indentical pairs of public/private keys for different derivations paths like 44'/397'/0'/0'/0 44'/397'/0'/0'/1 44'/397'/0'/0'/2 and so on

```
getBlockChainDataFromMnemonic(
    mnemonic,
    passphrase = "",
    account = 0,
    change = 0,
    address = 0
  ) {
    const { CoinType, HDWallet, Base64, HexCoding } = window.WalletCore;
    const wallet = HDWallet.createWithMnemonic(mnemonic, passphrase);
    const privateKey = wallet.getDerivedKey(
      CoinType.near,
      account,
      change,
      address
    );
    const privateKeyString = Base64.encode(privateKey.data());
    const hexString = HexCoding.encode(privateKey.getPublicKeyEd25519().data());

    return JSON.stringify({
      mnemonic: wallet.mnemonic(),
      publicKey: hexString.substring(2),
      accountId: hexString.substring(2),
      privateKey: privateKeyString,
      passphrase: passphrase,
      derivationPath: {
        purpose: "44",
        coinType: "397",
        accountNumber: account,
        change: change,
        address: address,
      },
    });
  }
```
![image_2024-02-25_230329590](https://github.com/trustwallet/wallet-core/assets/123000776/df4dda3f-92e0-42c0-ac7e-93ba43841ecc)

**Comment by satoshiotomakan** [comment, marked as answer]
Hi @ArturLevchuk, thank you for opening the issue!
NEAR uses ED25519 cryptography that doesn't support non-hardened child keys as per standard BIP32.
Unfortunately, `HDWallet::getDerivedKey` consider change and address indexes as non-hardened always, so it does not work properly with blockchains built upon ED25519.
Please consider using `HDWallet::getKey` with the derivation path as a string where all indexes are hardened (ends with `'`):
```ts
const privateKey = wallet.getDerivedKey(CoinType.near, "m/44'/397'/1'/0'/2'");
```

**Reply by ArturLevchuk** [reply]
Yeah after using getKey it works! Thank you very much)
